[#30] refactor the whole project (#31)
dermatz/ddev-woodoo-buildtools-magento: README.md

---

> This ddev add-on helps you to manage your theme build processes in Magento total easy. You can run a build command to build all your themes or a specific theme. Don't matter if the theme is a Magento Standard or Custom Theme, or maybe a Hyva theme. It automatically takes care which type of theme you use and will build it for you. If a dependency is missing, Woodoo will install it.

Reviewer: ```suggestion
This ddev add-on helps you to manage your theme build processes in Magento easily. You can run a build command to build all your themes or a specific theme. It doesn't matter if the theme is a Magento Standard Theme, a Custom Theme, or a Hyva theme. It automatically takes care of which type of theme you use and builds it for you. If a dependency is missing, Woodoo will install it.
```

```shell
ddev frontend init
```

---

**3. Edit theme path in .ddev/config-themes.yaml**
Open `.ddev/config-themes.yaml` and specify relative path to the theme based on your project structure. (E.g. `src/themeVendor/theme ...` or `app/design/frontend/themeVendor/theme/ ...`)

## Update Woodoo to latest stable Version

If a newer Release is available you will be notified in you ClI about new versions. To update use `ddev frontend-update` for the latest stable Version.

```shell
  ddev frontend-update
```

## Get Developer Preview

If you want to get the latest developer preview (can be instable) use `ddev frontend-update -dev`

```shell
  ddev frontend-update -dev
```

## Usage

```shell
Usage: ddev frontend [command] [option or theme]

Command:
  init, install         Initial Woodoo Setup to read all themes from dataabase and write to .ddev/config-themes.yaml
  help, -help, -h       Displays this help page
  version, --version    Show version
  check, selftest       Frontend Self-Test
  frontend-update       Update to the latest Version (stable Version)

>   frontend-update -dev  Update to the latest Main-Branch Version (not recommented)

Reviewer: ```suggestion
  frontend-update -dev  Update to the latest Main-Branch Version (not recommended)
```

  themes                List all available themes that are configured in .ddev/config-themes.yaml
  build                 Builds all themes that are configured in .ddev/config-themes.yaml

>   build -f              Builds all configured themes without yes/no question

Reviewer: ```suggestion
  build -f              Builds all configured themes without yes/no confirmation
```

  build theme           Build a specific theme
  watch theme           Watch for CSS and JS changes in a specific theme

Option:

>   -f                    Force the build command to run without yes/no question

Reviewer: ```suggestion
  -f                    Force the build command to run without yes/no confirmation
```
